lib/trento/application/usecases/alerting/alerting.ex

```diff
@@ -0,0 +1,65 @@
+defmodule Trento.Application.UseCases.Alerting do
+  @moduledoc """
+  Provides a set of functions of Alerting related usecases.
+  """
+
+  alias Trento.{
+    ClusterReadModel,
+    DatabaseReadModel,
+    HostReadModel,
+    SapSystemReadModel
+  }
+
+  alias Trento.Infrastructure.Alerting.EmailAlert
```

Comment:
This breaks the dependency rule :smile: 
From application we're having a hard dependency from infrastructure.

I'd prefer a level of indirection here, yet maybe we can keep it for later.

Reply:
@nelsonkopliku this because emailalert is a presentation component, wrapping a template and so on, not an infrastructural component, so i think this is misplaced in the beginning.
you can move it to the usecase folder i believe

Reply:
I see your point, and considering that it makes sense to have `EmailAlert` in application layer.
Changed like this
![image](https://user-images.githubusercontent.com/8167114/161534411-7382a405-f77d-4286-8397-8874178fc20c.png)

Good for me.

I want to share my point of view.

I still see emails as an infrastructural thing.
Why? Because to me email is a mean of notifying someone about something.
So I see the email as _how the notification gets delivered_

Some notifications could be email notifications, other ones might SMS, and others might be integration with other systems... and so forth.

So the concepts I see are (forgive the naming :smile: ) 
- `Notification`: what needs to be notified 
- `Transport`: how the notification is delivered

I see how this is **not what we currently need**, just sharing also for future discussions.